### PR TITLE
move layer tree code to specific folder

### PR DESCRIPTION
**Pull request by 3nids**
I used the same approach than QGIS by using include_directories. This makes it a bit easier to include files but maybe at the sake of clarity (on where each include belongs). Thoughts?

### CODE REVIEW

File: kadas/gui/milx/kadasmilxitem.cpp [hunk 1]
```diff
@@ -33,7 +34,6 @@
 
 #include <kadas/gui/milx/kadasmilxitem.h>
 #include <kadas/gui/milx/kadasmilxlayer.h>
-#include <kadas/app/kadasapplication.h>
```
Comment (3nids) [inline kadas/gui/milx/kadasmilxitem.cpp:36]: This fixes this include to an app header from gui.

**Comment by m-kuhn** [comment]
No strong opinion, alignment with QGIS sounds good.
While on this, I think we should strive towards C++ best practices with `"double-quoted"` includes for local deps and `<angle-bracket>` include for external deps

**Comment by 3nids** [comment]
partially done

File: kadas/app/CMakeLists.txt [hunk 2]
```diff
@@ -87,6 +95,7 @@ FILE(GLOB kadas_HDR
   bullseye/*.h
   guidegrid/*.h
   iamauth/*.h
+  layertree/*.h
```
Comment (m-kuhn) [inline kadas/app/CMakeLists.txt:98]: Note: at some point we should also make all headers explicitly listed, but it's a minor issue only.

**Review by m-kuhn — APPROVED** [review]
Let's merge soon to avoid conflicts

**Pull request merged**